src/rust/engine/fs/store/src/lib.rs

```diff
@@ -410,14 +414,18 @@ impl Store {
     initial_lease: bool,
   ) -> Result<Digest, String> {
     let local = self.local.clone();
-    local
+    let digest = local
       .store_bytes(
         EntryType::Directory,
         None,
         directory.to_bytes(),
         initial_lease,
       )
-      .await
+      .await?;
+    if cfg!(debug_assertions) {
+      protos::verify_directory_canonical(digest, directory)?;
```

Comment:
Maybe use `expect` instead of `?`?

Reply:
My bias is definitely toward not panicing... I primarily `unwrap`'d in the other case since `walk` is infallible, and it was easier than setting a mutable error or something.